File: js/tests/circle.test.js
const Circle = require("../circle");

const round = (number) => {
  let factor = Math.pow(10, 5);

  return Math.round(number * factor) / factor;
};

describe("Circle", () => {
  describe("getArea", () => {
    describe("Returns area of circle from radius input", () => {
      const data = [
        [20, 1256.63706],
        [2, 12.56637],
        [4.4, 60.82123]
      ];

      test.each(data)("new Circle(%p).getArea() = %p", (radius, result) => {
        let circle = new Circle(radius);

        let actual = round(circle.getArea());

        expect(actual).toEqual(result);
      });
    });
  });

  describe("getPerimeter", () => {
    describe("Return circumference of circle from radius input", () => {
      const data = [
        [20, 125.66371],
        [2, 12.56637],
        [4.4, 27.64602]
      ];

      test.each(data)(
        "new Circle(%p).getPerimeter() = %p",
        (radius, result) => {
          let circle = new Circle(radius);

          let actual = round(circle.getPerimeter());

          expect(actual).toEqual(result);
        }
      );
    });
  });
});

File: CSharp/Tests/CircleTest.cs
using System;
using Xunit;

namespace CSharp.Tests
{
    public class CircleTest
    {
        [Theory]
        [InlineData(20, 1256.63706)]
        [InlineData(2, 12.56637)]
        [InlineData(4.4, 60.82123)]
        public void GetArea_DoubleRadiusValue_ReturnAreaOfCircle(double radius, double expected)
        {
            var circle = GenerateCircle(radius);

            var actual = Round(circle.GetArea());

            Assert.Equal(expected, actual);
        }

        [Theory]
        [InlineData(20, 125.66371)]
        [InlineData(2, 12.56637)]
        [InlineData(4.4, 27.64602)]
        public void GetPerimeter_DoubleRadiusValue_ReturnCircumference(double radius, double expected)
        {
            var circle = GenerateCircle(radius);

            var actual = Round(circle.GetPerimeter());

            Assert.Equal(expected, actual);
        }

        public Circle GenerateCircle(double radius) => new Circle(radius);

        private double Round(double number)
        {
            double factor = Math.Pow(10, 5);

            return Math.Round(number * factor) / factor;
        }
    }
}